engines/groovie/t11hgame.cpp

```diff
@@ -0,0 +1,251 @@
+	// FIXME: properly implement Beehive game
+	// for now just auto-solve the puzzle so the player can continue
+	_scriptVariables[13] = 5;
+}
+
+
+void T11hGame::opPente() {
+	// FIXME: properly implement Pente game (the final puzzle)
+	// for now just auto-solve the puzzle so the player can continue
+	_scriptVariables[5] = 4;
+}
+
+
+/*
+* Puzzle in the Gallery.
+* The aim is to select the last part of the image. When selecting a part all surrounding parts are also selected
+* +--------------------+--------------------------------+--------+
+* |         1/1A       |       2/1B                     |        |
+* |  +--------------+--+--------------------------+-----+        |
+* |  |              |                             |              |
+* +--+     4/1D     |            5/1E             |       3/1C   |
+* |                 |                             |              |
+* +-----+--------+--+--------+-----------------+--+--------+     |
+* |     |        |           |                 |           |     |
+* |     |        |           |                 |           |     |
+* |     |        |   8/21    |                 |           |     |
+* |     |        |           |     +-----------+           |     |
+* |     |        |           |     |           |           |     |
+* |     |        +-----------+     |   10/23   |   9/22    |     |
+* |     |                          |           |           |     |
+* |     |           7/20           +-----+-----+           +-----+
+* |     |                          |     |     |           |     |
+* |     +--------------------------+     |     |           |     |
+* |              6/1F                    |     |           |     |
+* +-----------+-----------+-----+--+     | 11  |           | 12  |
+* |   13/26   |           |     |  |     | /   |           | /   |
+* |     +-----+-----+     |     |  |     | 24  +-----------+ 25  |
+* |     |           |     |     |  |     |     |           |     |
+* +-----+   17/2A   |     |     |16|     |     |           |     |
+* |     |           |     |     |/ |     |     |           |     |
+* |     +-----+-----+     |     |29|     |     |           +-----+
+* |           |           |     |  |     |     |           |     |
+* |           |           |     |  |     +-----+   18/2B   |     |
+* |   19/2C   |   14/27   |     |  |           |           |     |
+* |           |           |     |  +-----------+           |     |
+* |           |           |     |  |           |           |     |
+* |           |           |     +--+   15/28   |           |     |
+* |           |           |                    |           |     |
+* |           +--------+--+--------------------+-----------+     |
+* |           | 20/2D  |              21/2E                      |
+* +-----------+--------+-----------------------------------------+
+*/
+void T11hGame::opGallery() {
+
+	byte field1[21];
+	byte field2[21];
+	byte var_18[21];
+	int var_1c, eax, edx, ecx;

```

Comment:
Is there a way to name these at least `varXX` or better give them more meaningful names?

Reply:
none of this code does anything yet so it's hard to say currently, maybe when the reverse engineering on it is completed

it could all be commented out right up to `int selectedPart;`